### PR TITLE
CosmosDB: fixing the validation on the name field

### PR DESCRIPTION
**Pull request by tombuildsstuff**
Following on from #262 - this fixes the validation to ensure upper-case characters are rejected in the `plan` phase rather than the `apply` phase.

### CODE REVIEW

File: azurerm/resource_arm_cosmos_db_account.go [hunk 1]
```diff
@@ -398,7 +398,7 @@ func resourceAzureRMCosmosDBAccountFailoverPolicyHash(v interface{}) int {
 func validateAzureRmCosmosDBAccountName(v interface{}, k string) (ws []string, errors []error) {
 	value := v.(string)
 
-	r, _ := regexp.Compile("[a-z0-9-]")
+	r, _ := regexp.Compile("^([a-z0-9\\-]+)$")
```
Comment (radeksimko) [inline azurerm/resource_arm_cosmos_db_account.go]: I can believe the escaping & quantifier (`+`) is necessary, but are the control characters like brackets & `^` & `$` too?
I mean, it applies to the whole string anyway and you don't need to capture anything via subgroup here?

**Review by radeksimko — APPROVED** [review]
The expression does no harm and does what it's supposed to do, so LGTM, but can be less verbose, IMO.

File: azurerm/resource_arm_cosmos_db_account.go (hunk 1 above)
Comment (tombuildsstuff) [inline azurerm/resource_arm_cosmos_db_account.go]: Running the tests it looks like we can remove the brackets but not the `^`&`$` - updated

**Comment by tombuildsstuff** [comment]
Tests pass:
```
$ acctests azurerm TestAccAzureRMCosmosDBAccountName_validation
=== RUN   TestAccAzureRMCosmosDBAccountName_validation
--- PASS: TestAccAzureRMCosmosDBAccountName_validation (0.00s)
PASS
ok  	github.com/terraform-providers/terraform-provider-azurerm/azurerm	0.017s
```

File: azurerm/resource_arm_cosmos_db_account.go (hunk 1 above)
Comment (radeksimko) [inline azurerm/resource_arm_cosmos_db_account.go]: Apparently I have still a lot to learn about regular expressions! 😅

**Pull request merged**